### DISCUSSION TITLE
Is it possible to restrict inclusion of plugin scripts?

### DISCUSSION THREAD

**Discussion by BsNoSi** [datenstrom/community#254] [Ask a question] [answered]
Is it possible to restrict the header link to plugin scripts to pages, where the plugin is used?

currently, all pages hold a script link

```
<script type="text/javascript" src="…"></script>
```

in their header, if a plugin is installed, that requires a script. If the plugin is used only on subpages, this delayes – with no need – the load time of the homepage and others.

**Comment by markseuffert** [comment, marked as answer]
Sorry, not that I know. I remember that we looked into this and couldn't detect reliably when a plugin needs to include extra headers and when not. Now that there are many plugins, with many CSS and JavaScript resources, maybe it's time to find a better design. Do you have any problems?

**Comment by BsNoSi** [comment]
Sometimes external scripts take a long load time, that slows down the loading speed of the homepage. 

Additionally, there is a script load with no need for it, which is relevant on mobile devices (with volume restrictions, costs).

I admit, that this is pain at a very high level.

**Comment by zorae** [comment]
1. Copy the <script… lines the plugin puts in your site header and save them to a file.
2. Edit /system/plugins/name_of_plugin.php and comment out the corresponding lines: `$output .= "<script…`
3. For pages that need the plugin, insert the snippet from step 1 into the page text. You can just do this in the web editor and Yellow will add the script tags inline.

**Comment by BsNoSi** [comment]
I read a Google recommendation to load large scripts from the footer. Then the visible part of the page is already loaded and can be read while the script is still loading.

Is there a way for plugins to insert a script into the footer instead of the header?

This should be possible either, because for a slider at top of page a script may be loaded of course previously. This has to be a desicion of plugin designer, if it simply slows down start of the plugin or influence the presentation. Maybe a plugin can have a `config.ini` switch, to load from header or footer.

**Comment by schulle4u** [comment]
You can already do this using the `onExtra()`function from the API. 

```
// Handle page extra HTML data
function onExtra($name)
{
	$output = NULL;
	if($name=="footer")
	{
		$output .= "Your script goes here\n";
	}
	return $output;
	}
```
